ledger/src/blockstore.rs

```diff
@@ -10979,6 +10979,7 @@ pub mod tests {
                             let slot = shreds[0].slot();
                             // Grab this lock to block `get_slot_entries` before it fetches completed datasets
                             // and then mark the slot as dead, but full, by inserting carefully crafted shreds.
+                            #[allow(clippy::readonly_write_lock)]
```

Comment:
This one is kind of strange, the lock being obtained is specifically marked as unused by naming it `_lowest_cleanup_slot`. So, I feel like clippy shouldn't be complaining at us about read vs. write ... might go poke around the clippy repo / update my compiler and repro locally

Reply:
yeah. it may be a bug in clippy but could we use this one as a stopgap and have another issue for tracking? 🥹

Reply:
I didn't get the chance to dig around, can you please add a comment:
```rust
// Possible clippy bug, the lock is unused so clippy lint shouldn't care about read vs. write lock
```

Reply:
okay! I added it in the same line so as not to distract from the lock logic itself: [288a01b](https://github.com/anza-xyz/agave/pull/1336/commits/288a01bf0cbf7d9e558f4e4c61b0e024128755ed)

Reply:
I don't think this is a clippy bug.

well, WriteLockGuard is unused but the lock is happening and seem to be needed to create a critical section.

Reply:
Per the clippy lint page [here](https://rust-lang.github.io/rust-clippy/master/index.html#/readonly_write_lock):
> Looks for calls to RwLock::write where the lock is only used for reading.

I agree that the lock is seemingly necessary. However, given that the lock is unused, it is not clear to me why clippy would suggest that a read only would be acceptable over a write lock. I would expect the lint to occur in a situation like this:
```rust
let lock = RwLock::new(10);
let w_lock = lock.write().unwrap();
println!("The read value of lock is {}", w_lock);
drop(w_lock);
```
But again, there is no usage of the lock to suggest that a read lock would suffice over a write lock